src/js/tools/drawing/DitheringTool.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @provide pskl.tools.drawing.DitheringTool
+ *
+ * @require pskl.utils
+ */
+(function() {
+  var ns = $.namespace('pskl.tools.drawing');
+
+  ns.DitheringTool = function() {
+    ns.SimplePen.call(this);
+    this.toolId = 'tool-dithering';
+    this.helpText = 'Dithering tool';
+  };
+  pskl.utils.inherit(ns.DitheringTool, ns.SimplePen);
+
+  /**
+   * @override
+   */
+  ns.DitheringTool.prototype.applyToolAt = function(col, row, color, frame, overlay, event) {
+    // Use primary selected color on cell with either an odd col or row.
+    // Use secondary color otherwise.
+    // When using the right mouse button, invert the above behavior to allow quick corrections.
+    var usePrimaryColor = (col + row) % 2;
+    var invertColors = event.button === Constants.RIGHT_BUTTON;
```

Comment:
We cannot rely on event.button here, because applyToolAt can be triggered by a mousemove event, which will not have the expected 'button' value. (see comment in DrawingController getCurrentColor_).

The only way to know if we are on the right button is to check the color here. Otherwise we could modify the event object passed by the DrawingController to the tools